### ISSUE TITLE
[Bug]: Milvus is crashed with panic "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#19939]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master 2ec1c0816
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus==2.2.0.dev57
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Milvus is crashed with panic "invalid memory address or nil pointer dereference"

```
ci.var.log.containers.md-19937-1-pr-milvus-indexcoord-78b4b5f94d-vpm2r_milvus-ci_indexcoord-8a7cb47ef48054e6c40ce744c2dfccd737ab5fee3af4798abf7be195d6a5eb2f.log:2022-10-20T18:54:28.661480759+08:00 stderr F panic: All attempts results:
ci.var.log.containers.md-19937-1-pr-milvus-querycoord-858ddf66b7-n48cc_milvus-ci_querycoord-2c07702c77b04d68d5bb972207584b2b213c56494cbaf3f0ac6cb6c4643587cc.log:2022-10-20T18:54:29.192512294+08:00 stderr F panic: runtime error: invalid memory address or nil pointer dereference
ci.var.log.containers.md-19937-1-pr-milvus-rootcoord-5649d8544-lv5r6_milvus-ci_rootcoord-1d1dfd3f205274be503186ff00566dd6c6413bd026e05a5196f5a6a7b3c32c3a.log:2022-10-20T18:54:28.341245947+08:00 stderr F panic: runtime error: invalid memory address or nil pointer dereference
```

ci.var.log.containers.md-19937-1-pr-milvus-rootcoord-5649d8544-lv5r6_milvus-ci_rootcoord-1d1dfd3f205274be503186ff00566dd6c6413bd026e05a5196f5a6a7b3c32c3a.log
```
2022-10-20T18:54:28.338835798+08:00 stdout F [2022/10/20 10:54:28.338 +00:00] [ERROR] [roles/roles.go:307] ["Get signal to exit\n"] [signal=terminated] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:307\ngithub.com/milvus-io/milvus/cmd/milvus.(*run).execute\n\t/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:120\ngithub.com/milvus-io/milvus/cmd/milvus.RunMilvus\n\t/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60\nmain.main\n\t/go/src/github.com/milvus-io/milvus/cmd/main.go:26\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"]
2022-10-20T18:54:28.338890402+08:00 stdout F [2022/10/20 10:54:28.338 +00:00] [DEBUG] [rootcoord/service.go:276] ["Rootcoord stop"] [Address=10.105.1.71:53100]
2022-10-20T18:54:28.341245947+08:00 stderr F panic: runtime error: invalid memory address or nil pointer dereference
2022-10-20T18:54:28.34127211+08:00 stderr F [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x2de95dd]
2022-10-20T18:54:28.341280192+08:00 stderr F
2022-10-20T18:54:28.341289027+08:00 stderr F goroutine 1 [running]:
2022-10-20T18:54:28.341297112+08:00 stderr F github.com/milvus-io/milvus/internal/rootcoord.(*Core).Stop(0xc001048300)
2022-10-20T18:54:28.341321065+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:663 +0x5d
2022-10-20T18:54:28.341356322+08:00 stderr F github.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).Stop(0xc0011ea420)
2022-10-20T18:54:28.34136922+08:00 stderr F     /go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:301 +0x712
2022-10-20T18:54:28.341386128+08:00 stderr F github.com/milvus-io/milvus/cmd/components.(*RootCoord).Stop(0x6000103?)
2022-10-20T18:54:28.341388954+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/components/root_coord.go:67 +0x1d
2022-10-20T18:54:28.341400117+08:00 stderr F github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc0010f1e50, 0x0, {0x0, 0x0})
2022-10-20T18:54:28.341406287+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:311 +0xc3c
2022-10-20T18:54:28.341409447+08:00 stderr F github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc00051c330, {0xc000050090?, 0x3, 0x3}, 0xc000562120)
2022-10-20T18:54:28.341412473+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:120 +0x733
2022-10-20T18:54:28.341416119+08:00 stderr F github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc000050090?, 0x3, 0x3})
2022-10-20T18:54:28.341425525+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e
2022-10-20T18:54:28.341429104+08:00 stderr F main.main()
2022-10-20T18:54:28.341432569+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e
```
ci.var.log.containers.md-19937-1-pr-milvus-indexcoord-78b4b5f94d-vpm2r_milvus-ci_indexcoord-8a7cb47ef48054e6c40ce744c2dfccd737ab5fee3af4798abf7be195d6a5eb2f.log
```
2022-10-20T18:54:28.658310545+08:00 stdout F s.func1\n/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:37 github.com/milvus-io/milvus/internal/util/retry.Do\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:89 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:104 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy\n/go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:141 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).init\n/go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:81 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Run\n/go/src/github.com/milvus-io/milvus/cmd/components/index_coord.go:51 github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Run\n\n"]
2022-10-20T18:54:28.661480759+08:00 stderr F panic: All attempts results:
2022-10-20T18:54:28.661497355+08:00 stderr F attempt #1:err: find no available rootcoord, check rootcoord state
2022-10-20T18:54:28.661501176+08:00 stderr F , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
2022-10-20T18:54:28.66150628+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:304 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall
2022-10-20T18:54:28.661509568+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/client/client.go:125 github.com/milvus-io/milvus/internal/distributed/rootcoord/client.(*Client).GetComponentStates
2022-10-20T18:54:28.661513556+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:65 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1
2022-10-20T18:54:28.661517752+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:37 github.com/milvus-io/milvus/internal/util/retry.Do
2022-10-20T18:54:28.661521258+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:89 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates
2022-10-20T18:54:28.661524532+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:104 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy
2022-10-20T18:54:28.661529128+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:141 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).init
2022-10-20T18:54:28.661536177+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:81 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Run
2022-10-20T18:54:28.66153984+08:00 stderr F /go/src/github.com/milvus-io/milvus/cmd/components/index_coord.go:51 github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Run
2022-10-20T18:54:28.661542878+08:00 stderr F
2022-10-20T18:54:28.661545803+08:00 stderr F attempt #2:err: find no available rootcoord, check rootcoord state
2022-10-20T18:54:28.661549083+08:00 stderr F , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
2022-10-20T18:54:28.661551636+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:304 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall
2022-10-20T18:54:28.661553999+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/client/client.go:125 github.com/milvus-io/milvus/internal/distributed/rootcoord/client.(*Client).GetComponentStates
2022-10-20T18:54:28.661556416+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:65 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1
2022-10-20T18:54:28.661559166+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:37 github.com/milvus-io/milvus/internal/util/retry.Do
2022-10-20T18:54:28.661561625+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:89 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates
2022-10-20T18:54:28.661563995+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:104 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy
2022-10-20T18:54:28.661566234+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:141 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).init
2022-10-20T18:54:28.661569821+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:81 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Run
2022-10-20T18:54:28.661572914+08:00 stderr F /go/src/github.com/milvus-io/milvus/cmd/components/index_coord.go:51 github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Run
2022-10-20T18:54:28.661575166+08:00 stderr F
2022-10-20T18:54:28.661577474+08:00 stderr F attempt #3:err: find no available rootcoord, check rootcoord state
2022-10-20T18:54:28.661580007+08:00 stderr F , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
2022-10-20T18:54:28.661582813+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:304 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall
2022-10-20T18:54:28.661585383+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/client/client.go:125 github.com/milvus-io/milvus/internal/distributed/rootcoord/client.(*Client).GetComponentStates
```

### Expected Behavior

No error

### Steps To Reproduce

```markdown
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-19937/1/pipeline
```

### Milvus Log

[artifacts-milvus-distributed-PR-19937-1-pymilvus-e2e-logs.tar.gz](https://github.com/milvus-io/milvus/files/9829546/artifacts-milvus-distributed-PR-19937-1-pymilvus-e2e-logs.tar.gz)

**Comment by longjiquan** [comment]
Will be fixed by #20019 . Please help to check this, @binbinlv , thx.
/assign @binbinlv 
/unassign

**Comment by binbinlv** [comment]
Fixed and closed in recently CI and nightly.
